src/other/broadcasting.jl

```diff
@@ -0,0 +1,72 @@
+struct LazyNewColDataFrame
+    df::DataFrame
+    col::Symbol
+end
+
+# we allow LazyNewColDataFrame only for data frames with at least one column
+Base.axes(x::LazyNewColDataFrame) = axes(x.df[1])
+
+Base.maybeview(df::AbstractDataFrame, idxs...) = view(df, idxs...)
+
+function Base.maybeview(df::AbstractDataFrame, idxs)
+    if ncol(df) == 0
+        throw(ArgumentError("Broadcasting into a data frame with no columns is not allowed"))
+    end
+    if idxs isa Symbol
+        if !haskey(df, idxs)
+            if !(df isa DataFrame)
+                # this will throw an appropriate error message
+                df[idxs]
+            end
+            return LazyNewColDataFrame(df, idxs)
+        end
+    end
+    view(df, idxs)
+end
+
+function Base.copyto!(lazydf::LazyNewColDataFrame, bc::Base.Broadcast.Broadcasted)
+    if isempty(lazydf.df)
+        throw(ArgumentError("creating a column via broadcasting is not allowed on empty data frames"))
+    end
+    T = mapreduce(i -> typeof(bc[i]), promote_type, eachindex(bc); init=Union{})
```

Comment:
This is tricky because `bc` can have `Any` eltype (or in general - some abstract eltype). I could use `[v for v in bc_column]` style, and it would do automatic type promotion. However, there is a problem with `CategoricalArrays`, as if a column is categorical a comprehension will not create a proper container type (as you know very well 😄).

So - in summary - if we accept that we do not produce a narrowest possible type for the column this part of code can be dropped.

@nalimilan Please let me know which approach you would prefer.

In an ideal world I think that compiler should optimize out this `mapreduce` if it knows that `bc` has a homogeneous concrete eltype, but I have not checked if it happens.

Reply:
Ah, I guess `bc` could have eltype `Any` when calling e.g. `f.(x)`. Then that approach is fine. Though maybe check that the compiler optimizes this out when it has a concrete eltype, and if not add a check to skip it manually.

Reply:
OK - I will have look at it.